src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -524,25 +526,35 @@ private bool TryGetUnmanagedCallingConventionFromModOpt(MethodSignature signatur
                 if (defType.Namespace != "System.Runtime.CompilerServices")
                     continue;
 
-                // Take the first recognized calling convention in metadata.
+                // Look for a recognized calling convention in metadata.
                 switch (defType.Name)
                 {
                     case "CallConvCdecl":
-                        callConv = CorInfoCallConv.CORINFO_CALLCONV_C;
-                        return true;
+                        SetCallConvLocal(CorInfoCallConv.CORINFO_CALLCONV_C, ref callConvLocal);
```

Comment:
Nit: Not a big fan of the local function. It creates a capture object with an unnecessary allocation and I'm not sure if this improves readability over something like:

```csharp
CorInfoCallConv? callconvLocal = defType.Name switch
{
    "CallConvCdecl" => CorInfoCallConv.CORINFO_CALLCONV_C,
    // ...
    _ => null
};

if (callConvLocal.HasValue)
{
    if (found)
        throw...
    callConv = callConvLocal.Value;
    found = true;
}

Reply:
I'm honestly not a huge fan either. Happy to change it to what people think would be better.